flybywiresim/a32nx
# feat(fms): handle non-msfs approach transition coding

Fixes #[issue_no]

## Summary of Changes

Navigraph does not separate approach transitions out as expected by MSFS, and as a consequence many transitions are not available when using their navdata. We now find IAFs embedded within other transitions (as allowed by ARINC424, not MSFS), and append them to the list of available transitions. These transitions cannot be synced to the MSFS flight plan system when selected as they do not exist there, but that is better than not having them available at all. Let's hope Navigraph will rethink this decision in future.

Also fixed a couple of small formatting issues on the page while I was there (it now shows "NO VIA" instead of "NO VIAS", and the selected VIA is highlighted in green.

## Screenshots (if necessary)


Before with Navigraph data, KIAD R19CY with DIMKE transition missing:
![image](https://user-images.githubusercontent.com/9995998/227712058-2ff38a6a-3dba-4080-9223-59da850afae4.png)

After:
![image](https://user-images.githubusercontent.com/9995998/227711661-aaeb3273-9154-4657-8fac-fbfacb6c3638.png)

Default MSFS data (CRVER transition is missing entirely from the Navigraph/Jeppesen data...):
![image](https://user-images.githubusercontent.com/9995998/227713506-1ca2acf9-ea67-4c8f-9250-ebbcef31f20e.png)
![image](https://user-images.githubusercontent.com/9995998/227714237-2dac1dd2-4340-4b0a-8a52-747fa34e9ce8.png)


## References





## Additional context



Discord username (if different from GitHub):

## Testing instructions

- With Navigraph data installed, check an airport that was missing some approach transitions before, and see if they are now present.
- Check that approach transition page scrolling works
- Check that approach transitions are strung into the flightplan when selected


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
